src/lib.ml

```diff
@@ -819,20 +814,10 @@ and resolve_user_deps db deps ~allow_private_deps ~pps ~stack =
         closure_with_overlap_checks None pps ~stack
       in
       let deps =
-        let rec check_runtime_deps acc pps = function
-          | [] -> loop acc pps
-          | lib :: ppx_rts ->
-            check_private_deps ~lib ~loc ~allow_private_deps >>= fun rt ->
-            check_runtime_deps (rt :: acc) pps ppx_rts
-        and loop acc = function
-          | [] -> Ok acc
-          | pp :: pps ->
-            pp.ppx_runtime_deps >>= fun rt_deps ->
-            check_runtime_deps acc pps rt_deps
-        in
         deps >>= fun deps ->
-        pps  >>= fun pps  ->
-        loop deps pps
+        pps >>= Result.concat_map ~f:(fun pp -> pp.ppx_runtime_deps)
+        >>| List.rev_append deps
+        >>= Result.List.map ~f:(check_private_deps ~loc ~allow_private_deps)
```

Comment:
This is not the same as before: `check_private_deps` is now called a second time for each element in `deps`. It's also reversing `deps`. It shouldn't matter, but sometimes such changes have unexpected effects on user code.

Reply:
I reverted this change because maintaining the order is too hard here. I also don't even preserve the order of the ppx rt deps, so if the order needs to be the same, there's no way to salvage this change.